type.go

```diff
@@ -539,6 +539,78 @@ func (typeStruct) Generate(r *rand.Rand, size int) reflect.Value {
 	return reflect.ValueOf(typeStruct(typeStructFields))
 }
 
+type typeList struct {
+	Type Type
+	Size uint64
+}
+
+func (typeList) Kind() Kind {
+	return KindList
+}
+
+func (t typeList) UnmarshalValue(buf []byte, rem int) (Value, []byte, int, error) {
```

Comment:
First, you should unmarshal the length of the list (you can use `surge.UnmarshalLen` or `surge.UnmarshalU32`). After that, you know how many elements of type `t.Type` to unmarshal.